ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -71,23 +71,16 @@ private[kvutils] class TransactionCommitter(
     "validate_contract_keys" -> validateContractKeys,
     "validate_model_conformance" -> validateModelConformance,
     "blind" -> blind,
-    "remove_unnecessary_nodes" -> removeUnnecessaryNodes,
+    "trim_unnecessary_nodes" -> trimUnnecessaryNodes,
     "build_final_log_entry" -> buildFinalLogEntry,
   )
 
-  private def contractIsActiveAndVisibleToSubmitter(
+  private def contractIsActive(
```

Comment:
Can you give a bit of color as why we no longer need to check for visibility?

Reply:
This is quite subtle and articulated and it has to do with the (limited) power of this check and with the evolution of privacy. I recommend to refer to design comments.

Reply:
When we wrote the multi-party submission design document, we realized that the ledger should not have done the visibility check in the first place.

Note for example that post-commit validation in sandbox classic (e.g., `ActiveLedgerState.lookupContractLet`) doesn't do visibility checks either.

Reply:
I see, thanks. 🙇🏻